src/screens/Settings/HSwitchProfile/ModifyTwoFaSettings.res

```diff
@@ -0,0 +1,188 @@
+let h2TextStyle = HSwitchUtils.getTextClass((H2, Optional))
+let p2Regular = HSwitchUtils.getTextClass((P2, Regular))
+let p3Regular = HSwitchUtils.getTextClass((P3, Regular))
+
+module ResetTotp = {
+  @react.component
+  let make = (~checkStatusResponse) => {
+    open LogicUtils
+
+    let showToast = ToastState.useShowToast()
+    // let fetchDetails = APIUtils.useGetMethod()
+    // let verifyTotpLogic = TotpHooks.useVerifyTotp()
+    let (showVerifyModal, setShowVerifyModal) = React.useState(_ => false)
+    let (otpInModal, setOtpInModal) = React.useState(_ => "")
+    let (otp, setOtp) = React.useState(_ => "")
+    let (buttonState, setButtonState) = React.useState(_ => Button.Normal)
+    let (totpUrl, setTotpUrl) = React.useState(_ => "")
+
+    let generateNewSecret = async () => {
+      try {
+        // TODO: add generate new secret api
+        // let url = getURL(~entityName=USERS, ~userType=#RESET_TOTP, ~methodType=Get, ())
+        // let res =fetchDetails(url)
+        let secretObj =
+          [
+            ("secret", "aerf"->JSON.Encode.string),
+            ("totp_url", "otpauth:wefrq3"->JSON.Encode.string),
+          ]
+          ->Dict.fromArray
+          ->JSON.Encode.object
+        let res = [("secret", secretObj)]->Dict.fromArray->JSON.Encode.object
+
+        setTotpUrl(_ =>
+          res->getDictFromJsonObject->getDictfromDict("secret")->getString("totp_url", "")
+        )
+      } catch {
+      | _ => {
+          setOtp(_ => "")
+          setButtonState(_ => Button.Normal)
+        }
+      }
+    }
+
+    let verifyTOTP = async (~fromModal, ~methodType) => {
+      try {
+        setButtonState(_ => Button.Loading)
+        if otpInModal->String.length > 0 || otp->String.length > 0 {
+          // let body = [("totp", otp->JSON.Encode.string)]->getJsonFromArrayOfJson
+
+          // let _ = await verifyTotpLogic(body, methodType)
+          if fromModal {
+            setShowVerifyModal(_ => false)
+            generateNewSecret()->ignore
+          } else {
+            showToast(~message="Successfully reset the totp !", ~toastType=ToastSuccess, ())
+            RescriptReactRouter.push(
+              HSwitchGlobalVars.appendDashboardPath(~url="/account-settings/profile"),
+            )
+          }
+        } else {
+          showToast(~message="OTP field cannot be empty!", ~toastType=ToastError, ())
+        }
+        setButtonState(_ => Button.Normal)
+      } catch {
+      | _ => {
+          setOtpInModal(_ => "")
+          setOtp(_ => "")
+          setButtonState(_ => Button.Normal)
+        }
+      }
+    }
+
+    React.useEffect0(() => {
+      if (
+        checkStatusResponse->getBool("totp", false) ||
+          checkStatusResponse->getBool("recovery_code", false)
+      ) {
+        generateNewSecret()->ignore
+      } else {
+        setShowVerifyModal(_ => true)
+      }
+      None
+    })
+
+    <div>
+      <Modal
+        modalHeading="Verify OTP"
```

Comment:
It cannot always be Verify Totp.we need to allow user to enter recovery code also

Reply:
Added the feature to add access code as well here